projects/rocprofiler-systems/source/lib/rocprof-sys/library/rocprofiler-sdk.cpp

```diff
@@ -906,16 +990,224 @@ get_kernel_dispatch_timestamps()
     return _v;
 }
 
+#if(ROCPROFILER_VERSION >= 600)
+// To handle events without finalization, perfetto push must occur in start
+// Allows capture of worker thread implicit and sync tasks
+void
+ompt_tracing_callback_start(rocprofiler_callback_tracing_record_t record,
+                            rocprofiler_user_data_t* /*user_data*/,
+                            rocprofiler_timestamp_t ts)
+{
+    std::string_view _name;
```

Comment:
```suggestion
    static bool first_ompt_implicit_task = true;

    // Ignore first ompt_implicit_call as this is created after runtime initialization but
    // before first region. Also, the corresponding end event is not received because it ends
    // the application exits.
    if(first_ompt_implicit_task && (record.kind == ROCPROFILER_CALLBACK_TRACING_OMPT &&
                                  record.operation == ROCPROFILER_OMPT_ID_implicit_task))
    {
        first_ompt_implicit_task = false;
        return;
    }
    
    std::string_view _name;
```
Now that you have an explicit ompt handler, you can move this logic to here.